pfi.c

```diff
@@ -26,65 +30,132 @@ static char iface[IFNAMSIZ] = "eth0";
 module_param_string(interface, iface, IFNAMSIZ, 0644);
 MODULE_PARM_DESC(interface, "Interface to disable when interrupt was received.");
 
+static struct class *pfi_class;
+static struct device *pfi_device = NULL;
```

Comment:
Yeah, there's a reason. I copied them from different sources. ;-)

Actually, I probably didn't.

I think at one point the code I had possibly called `device_destroy` without necessarily successfully calling `device_create_file` and `device_destroy` properly ignores a NULL device. I don't think that's any longer a possibility, but these should be consistent.